From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@sourcemage.org>
Date: Sat, 22 Aug 2026 12:00:00 +0200
Subject: [PATCH] Include <string.h> in getopt.c

getopt.c only includes <string.h> when __GNU_LIBRARY__ is defined, which
no libc but glibc defines; on musl strcmp() and strlen() end up implicitly
declared, which GCC 14 and later reject.

Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 lib-src/getopt.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib-src/getopt.c b/lib-src/getopt.c
index d7719dc3888b..de884ac2b240 100644
--- a/lib-src/getopt.c
+++ b/lib-src/getopt.c
@@ -62,6 +62,8 @@
 
 #ifndef ELIDE_CODE
 
+#include <string.h>
+
 
 /* This needs to come after some library #include
    to get __GNU_LIBRARY__ defined.  */
